README.md

```diff
@@ -1,43 +1,103 @@
+Alibaba Cloud Load Balancer (SLB) Listener Terraform Module
+terraform-alicloud-slb-listener
+=====================================================================
 
-Terraform module which creates SLB HTTP, HTTPS, TCP or UDP Listener resources on Alibaba Cloud.
+English | [简体中文](https://github.com/terraform-alicloud-modules/terraform-alicloud-slb-listener/blob/master/README-CN.md)
 
-These types of resources are supported:
-
-* [SLB Listener](https://www.terraform.io/docs/providers/alicloud/r/slb_listener.html)
-
-Usage
------
-You can use this in your terraform template with the following steps.
-
-1. Adding a module resource to your template, e.g. main.tf
-
-
-         module "tf-slb" {
-            source = "terraform-alicloud-modules/slb-listener/alicloud"
-
-            slb = "lb-f893h8c4c"
+Terraform module which creates slb listener resources on Alibaba Cloud.
 
-            protocol = "http"
-            backend_port = 8080
-            frontend_port = 80
-
-            enable_health_check = true
-            health_check_connect_port = 80
-            health_check_uri = "www.alibabacloud.com"
-
-         }
-
-2. Setting values for the following variables through environment variables:
-
-    - ALICLOUD_ACCESS_KEY
-    - ALICLOUD_SECRET_KEY
+These types of resources are supported:
 
+* [Slb Listener](https://www.terraform.io/docs/providers/alicloud/r/slb_listener.html)
+
+## Terraform versions
+
+The Module requires Terraform 0.12 and Terraform Provider AliCloud 1.56.0+.
+
+## Usage
+
+```hcl
+module "slb_listener" {
+  source  = "terraform-alicloud-modules/slb-listener/alicloud"
+  profile = "Your-Profile-Name"
+  region  = "cn-beijing"
+  slb     = "lb-bp1xxxxxxxxxxxxxx"
+  listeners = [
+    {
+      server_group_ids  = "rsp-bp1xxxxxxx1,rsp-bp1xxxxxx2"
+      backend_port      = "80"
+      frontend_port     = "80"
+      protocol          = "tcp"
+      bandwidth         = "-1"
+      scheduler         = "wrr"
+      healthy_threshold = "4"
+      gzip              = "false"
+      health_check_type = "tcp"
+    }
+  ]
+  
+  // you can also leave this field unspecified to use the default value
+  health_check = {
+    health_check              = "on"
+    health_check_type         = "tcp"
+    healthy_threshold         = "3"
+    unhealthy_threshold       = "2"
+    health_check_timeout      = "5"
+    health_check_interval     = "2"
+    health_check_connect_port = "80"
+    health_check_uri          = "/"
+    health_check_http_code    = "http_2xx"
+  }
+  
+  // you can also leave this field unspecified to use the default value
+  advance_setting = {
+    sticky_session      = "on"
+    sticky_session_type = "server"
+    cookie_timeout      = "86400"
+    gzip                = "false"
+    retrive_slb_ip      = "true"
+    retrive_slb_id      = "false"
+    retrive_slb_proto   = "true"
+    persistence_timeout = "5"
+  }
+  
+  // you can also leave this field unspecified to use the default value
```

Comment:
x_forwarded_for will apply to all of listeners if it is not set in the listeners